test_runner/batch_others/test_proxy.py

```diff
@@ -22,6 +26,99 @@ def test_password_hack(static_proxy):
         static_proxy.safe_psql('select 1', sslsni=0, user=user, password=magic)
 
 
+@pytest.mark.asyncio
+async def test_psql_session_id(vanilla_pg, link_proxy):
+    """
+    Test copied and modified from: test_project_psql_link_auth test from cloud/tests_e2e/tests/test_project.py
+    This is only one half of the test: the half that establishes connection to the proxy and retrieves the session_id
+    The second half of the test required more heavy machinery to be ported from the cloud codebase
+        (such as: project_state(..), and psql_project_connect)
+    to connect to postgres and make sure the end-to-end behavior is satisfied.
+    """
+    psql = PSQL(
+        host=link_proxy.host,
+        port=link_proxy.proxy_port,
+    )
+    proc = await psql.run()
+
+    # Read line by line output of psql.
+    # We expect to see result in first 3 lines. [this is flexible, change if need be]
+    # Expected output:
+    #   A notice that contains a welcome to neon message that contains a http url in the form of
+    #   "http://dummy-uri/<psql_session_id>"
+    #   (here dummy-uri was passed from the neon_fixture that created the link_proxy
+    #   The url has to be on its own line.
+    #   Example:
+    #   "
+    #   NOTICE:  Welcome to Neon!
+    #   Authenticate by visiting:
+    #      http://dummy-uri/<psql_session_id>
+    #   "
+
+    max_num_lines_of_welcome_message = 15
+    attempt = 0
+    psql_session_id = None
+
+    log.info(f"Neon Welcome Message:")
+    for attempt in range(max_num_lines_of_welcome_message):
+        raw_line = await proc.stderr.readline()
+        if not raw_line:
+            # output ended
+            break
+        log.info(f"{raw_line}")
+        line = raw_line.decode("utf-8").strip()
+        if line.startswith("http"):
+            line_parts = line.split("/")
+            psql_session_id = line_parts[-1]
+            link_auth_uri = '/'.join(line_parts[:-1])
+            assert link_auth_uri == link_proxy.link_auth_uri, \
+                f"Line='{line}' should contain a http auth link of form '{link_proxy.link_auth_uri}/<psql_session_id>'."
```

Comment:
[urllib.parse](https://docs.python.org/3/library/urllib.parse.html#module-urllib.parse)

Reply:
the code is the same length either way and I kind of prefer without using urllib:

```
# using urlparse, 3 lines
url_parts = urlparse(line)
psql_session_id = url_parts.path[1:]
link_auth_uri = line[:-len(url_parts.path)]

# manual parsing, 3 lines
line_parts = line.split("/")
psql_session_id = line_parts[-1]
link_auth_uri = '/'.join(line_parts[:-1])
```